token.go

```diff
@@ -24,23 +24,29 @@ var DecodePaddingAllowed bool
 // using this package.
 var DecodeStrict bool
 
-// Keyfunc will be used by the Parse methods as a callback function to supply
+// KeyfuncFor will be used by the Parse methods as a callback function to supply
 // the key for verification.  The function receives the parsed, but unverified
 // Token.  This allows you to use properties in the Header of the token (such as
 // `kid`) to identify which key to use.
-type Keyfunc func(*Token) (interface{}, error)
+type KeyfuncFor[T Claims] func(*TokenFor[T]) (interface{}, error)
 
-// Token represents a JWT Token.  Different fields will be used depending on
+// Keyfunc is an alias for KeyfuncFor[Claims], for backward compatibility.
+type Keyfunc = KeyfuncFor[MapClaims]
+
+// TokenFor represents a JWT Token.  Different fields will be used depending on
 // whether you're creating or parsing/verifying a token.
-type Token struct {
+type TokenFor[T Claims] struct {
 	Raw       string                 // Raw contains the raw token.  Populated when you [Parse] a token
 	Method    SigningMethod          // Method is the signing method used or to be used
 	Header    map[string]interface{} // Header is the first segment of the token
-	Claims    Claims                 // Claims is the second segment of the token
+	Claims    T                      // Claims is the second segment of the token
 	Signature string                 // Signature is the  third segment of the token.  Populated when you Parse a token
 	Valid     bool                   // Valid specifies if the token is valid.  Populated when you Parse/Verify a token
 }
 
+// Token is an alias for TokenFor[Claims], for backward compatibility.
+type Token = TokenFor[MapClaims]
```

Comment:
We probably want something like this for even more compatibility:
```suggestion
type Token = TokenFor[Claims]
```